Error starting lein project: missing `clojure.tools.namespace` dep

## Expected

A new Leiningen project with a namespace requiring `donut.system.repl` should start without error.

## Actual

```
$ lein repl
...
Exception in thread "main" Syntax error macroexpanding at (donut/system/repl.clj:1:1).
        ...
Caused by: java.io.FileNotFoundException: Could not locate clojure/tools/namespace/repl__init.class, clojure/tools/namespace/repl.clj or clojure/tools/namespace/repl.cljc on classpath.
```

## Steps to Reproduce

1. Create a new lein project.
2. Add the donut.system dependency to `project.clj`.
   ```clj
   [club.donutpower/system "0.0.127"]
   ```
3. Require `donut.system.repl` in any namespace. 
   ```clj
   [donut.system.repl :as dsr]
   ```
4. Start a REPL.
   ```sh
   $ lein repl
   ```
5. Observe the `FileNotFoundException`.

## Versions

donut.system 0.0.127
Clojure 1.11.1

## More Info

The `clojure.tools.namespace` dependency is missing from production builds. This can be checked from the root of the donut.system project, observing no output:

```sh
$ clj -Stree | grep tools.namespace

```

This is a tricky one, because many developer and project configs include it as either an explicit or transitive dependency. And all the aliases in donut.system's `deps.edn` include it transitively:

```sh
$ clj -Stree -X:dev | grep tools.namespace
      . org.clojure/tools.namespace 0.2.11

$ clj -Stree -X:test | grep tools.namespace
  . org.clojure/tools.namespace 1.1.0

$ clj -Stree -X:test-cljs | grep tools.namespace
  . org.clojure/tools.namespace 1.0.0

$ clj -Stree -X:build | grep tools.namespace
    . org.clojure/tools.namespace 1.2.0
```

Tests pass! 😆

The issue is resolved by adding the dependency to `deps.edn`.

PR incoming.

--- a/deps.edn
+++ b/deps.edn
@@ -1,9 +1,10 @@
 {:paths ["src"]
- :deps  {org.clojure/clojure {:mvn/version "1.10.3"}
-         aysylu/loom         {:mvn/version "1.0.2"}
-         com.rpl/specter     {:mvn/version "1.1.3"}
-         metosin/malli       {:mvn/version "0.7.5"}
-         medley/medley       {:mvn/version "1.3.0"}}
+ :deps  {org.clojure/clojure         {:mvn/version "1.10.3"}
+         org.clojure/tools.namespace {:mvn/version "1.3.0"}
+         aysylu/loom                 {:mvn/version "1.0.2"}
+         com.rpl/specter             {:mvn/version "1.1.3"}
+         metosin/malli               {:mvn/version "0.7.5"}
+         medley/medley               {:mvn/version "1.3.0"}}
 
  :aliases
  {:dev

